src/mktag_kernel.c

```diff
@@ -0,0 +1,123 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+/*
+ * mktag_kernel.c - utility to write the tag_kernel file for TP Link Deco X20 jffs2 filesystem lookup
+ *
+ * Copyright (C) 2024 Damien Mascord <tusker@tusker.org>
+ */
+
+#include <arpa/inet.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdint.h>
+#include <unistd.h>
+#include <sys/stat.h>
+
+int fpread(void *buffer, size_t size, size_t nitems, size_t offset, FILE *fp);
+
+typedef struct __attribute__((scalar_storage_order("little-endian"))) _LINUX_FILE_TAG
+{
+	int32_t rootfsLen;
```

Comment:
This tool just writes the root fs length into the first 4 bytes and nulls the rest. I think we can do this better in a shell script.

Reply:
I will have a look into it, are there any endianess helpers so that the struct is written the correct way?